apps/dot_ext/scopes.py

```diff
@@ -22,10 +22,7 @@ def get_available_scopes(self, application=None, request=None, *args, **kwargs):
         Returns a list that contains all the capabilities related
         to the current application.
         """
-        if application:
-            return list(application.scope.values_list('slug', flat=True))
-        else:
-            return []
+        return list(ProtectedCapability.objects.values_list('slug', flat=True))
```

Comment:
This change is what allows the token flow to work if an application has no scopes associated with it in the DB. It does that by simply returning all `ProtectedCapability`s when asked what scopes the application has.

Reply:
I should say, after looking at the oauth2_provider scope interface, I think this change has the smallest blast radius.